internal/common/vote/vote.go

```diff
@@ -0,0 +1,188 @@
+package vote
+
+//will be generated from client
+var voteForRule map[int][]int
+var voteForElect map[int][]int
+
+func GetVotesForRule(ruleID string, numOfIslands int) map[int][]int {
+	votesLayoutRule := make(map[int][]int)
+	i := 1
+	for {
+		votesLayoutRule[i] = voteForRule[i]
+		i++
+		if i > numOfIslands {
+			break
+		}
+	}
+	return votesLayoutRule
+}
+
+func VoteRule(ruleID string, numOfIslands int) (int, int, bool, map[int][]int) {
+
+	//Get votes from islands for the rule changes.
+	//var votesLayoutRule map[int][]int
+	votesLayoutRule := GetVotesForRule(ruleID, numOfIslands)
+
+	//Calculate results of each island.
+	resultsOfAllIslands := make(map[int][]int)
+	j := 1
+	for {
+		if votesLayoutRule[j][0] > votesLayoutRule[j][1] {
+			resultsOfAllIslands[j] = []int{1, 0}
+		}
+		if votesLayoutRule[j][0] <= votesLayoutRule[j][1] {
+			resultsOfAllIslands[j] = []int{0, 1}
+		}
+		j++
+		if j > numOfIslands {
+			break
+		}
+	}
+
+	//Announce voting result
+	numacc := 0
+	numrej := 0
+	var ans bool
+	for _, v := range resultsOfAllIslands {
+		if v[0] == 1 {
+			numacc++
+		}
+		if v[1] == 1 {
+			numrej++
+		}
+	}
+	if numacc > numrej {
+		ans = true
+	}
+	if numacc <= numrej {
+		ans = false
+	}
+	return numacc, numrej, ans, votesLayoutRule
+}
+
+func GetVotesForElect(numOfIslands int) map[int][]int {
```

Comment:
Same for this, the implementation of the vote counting should be on the server side, and then the vote itself should come from the clients